src/couch_views/test/couch_views_indexer_test.erl

```diff
@@ -388,6 +389,77 @@ multipe_identical_keys_from_same_doc(Db) ->
         ], Out).
 
 
+fewer_multipe_identical_keys_from_same_doc(Db) ->
+    DDoc = create_ddoc(multi_emit_same),
+
+    Doc0 = couch_doc:from_json_obj({[
+        {<<"_id">>, list_to_binary(integer_to_list(0))},
+        {<<"val">>, 1},
+        {<<"extra">>, 3}
+    ]}),
+
+    {ok, _} = fabric2_db:update_doc(Db, DDoc, []),
+    {ok, {Pos, Rev}} = fabric2_db:update_doc(Db, Doc0, []),
+
+    {ok, Out1} = couch_views:query(
+            Db,
+            DDoc,
+            <<"map_fun1">>,
+            fun fold_fun/2,
+            [],
+            #mrargs{}
+        ),
+
+    ?assertEqual([
+            {row, [
```

Comment:
To reduce number of lines, could have a `row(Id, Key, Val)` so then it will look like

```
?assertEqual([
    row(<<"0">>, 1, 1),
    row(<<"0">>, 1, 2),
    row(<<"0">>, 1, 3)
], Out1).
```

Reply:
Done.